### DISCUSSION TITLE
Eloquent/Collection: Call to a member function getKey() on string

### DISCUSSION THREAD

**Discussion by LastDragon-ru** [laravel/framework#43016] [Q&A]
The following code works fine when collection contains models but will fail if `Collection` is empty :
```php
(new \Illuminate\Database\Eloquent\Collection())->map(fn() => 'converts item to string')->intersect(['a']);
```

Maybe `map()` it should return base collection if empty?

https://github.com/laravel/framework/blob/95ce178cbde9f9f38013862f677a5314da67af85/src/Illuminate/Database/Eloquent/Collection.php#L361-L363

_PS: Laravel v9.17.0_

**Comment by mattford** [comment]
Still the case in Laravel 12. 

Ran into this issue with mapWithKeys and I agree with you it should map into a base collection in the empty case, otherwise behaviour is different based on whether a query returned results or not. 

Also if the collection is empty I don't expect the Eloquent specific methods would be that useful.

However this would be a BC break for that reason I don't expect it to change, workaround is to call `toBase()` when you need to be absolutely sure it's a base collection
